docs/installation/integrations/next-js.md

```diff
@@ -0,0 +1,171 @@
+```
+
+The dependency should be visible in the `package.json` file as `ckeditor5-custom-build`. Additionally, we need an external dependency &ndash; the CKEditor&nbsp;5 React component. Next.js extends React, and hence, the component is valid here. You can install it using `npm install @ckeditor/ckeditor5-react` or the following command:
+
+```bash
+yarn add @ckeditor/ckeditor5-react
+```
+
+Next, we will use the installed dependencies in a React component. Create a new component in the components directory, for example, `components/custom-editor.js`. Inside the component file, import all necessary dependencies. Then, create a functional component that returns the CKEditor&nbsp;5 React component. The custom build ships with a default configuration, but you can customize it to your needs, as shown in the snippet below.
+
+```jsx
+// components/custom-editor.js
+
+import React from 'react';
+import { CKEditor } from "@ckeditor/ckeditor5-react";
+import Editor from "ckeditor5-custom-build";
+
+const editorConfiguration = {
+    toolbar: [
+        'heading',
+        '|',
+        'bold',
+        'italic',
+        'link',
+        'bulletedList',
+        'numberedList',
+        '|',
+        'outdent',
+        'indent',
+        '|',
+        'imageUpload',
+        'blockQuote',
+        'insertTable',
+        'mediaEmbed',
+        'undo',
+        'redo'
+    ]
+};
+
+function CustomEditor( props ) {
+		return (
+			<CKEditor
+				editor={ Editor }
+				config={ editorConfiguration }
+				data={ props.initialData }
+				onChange={ (event, editor ) => {
+					const data = editor.getData();
+					console.log( { event, editor, data } );
+				} }
+			/>
+		)
+}
+
+export default CustomEditor;
+```
+
+Our `CustomEditor` component is ready to be used inside a page. The usage will differ depending on the chosen routing strategy.
+
+You can create a second page inside the `pages` directory or replace the content of the existing `index.js` page &ndash; we will go with the second option. CKEditor&nbsp;5 is a client-side text editor and relies on the browser APIs, so we need to disable SSR for our custom component. We can lazily load the component using the `dynamic()` function built into Next.js.
+
+App Router, by default, uses server components. It means we need to mark a component as client-side explicitly. We can achieve that by using the `'use client'` directive at the top of a file, above your imports. Also, pay attention to the folder structure, which is different now. The `page.js` file should go into the `app` directory.
+
+```jsx
+'use client' // only in App Router
+
+import React from 'react';
+import dynamic from 'next/dynamic';
+
+const CustomEditor = dynamic( () => {
+  return import( '../components/custom-editor' );
+}, { ssr: false } );
+
+function Home() {
+  return (
+    <CustomEditor 
+      initialData='<h1>Hello from CKEditor in Next.js!</h1>'
+    />
+  );
+}
+
+export default Home;
+```
+
+### Page Router
```

Comment:
I would move this higher just after.

> First, put your unzipped custom build at the project root in the \`ckeditor5\` folder.

Unfortunately, this step is usually very unclear for devs (this is not popular nor best practice currently). And we need some visual guide there. If it's too long, we can cut out some next.js nested files/folders.